rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/RepositoryDatasetImpl.java

```diff
@@ -0,0 +1,192 @@
+			conn.add(statement);
+			conn.commit();
+		}
+	}
+
+
+	@Override
+	public boolean contains(Quad tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.hasStatement(statement, includeInferred);
+		}
+	}
+
+	@Override
+	public void remove(Quad tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(statement);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public void clear() {
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.clear();
+			conn.commit();
+		}
+	}
+
+	@Override
+	public long size() {
+		if (includeInferred) { 
+			// We'll need to count them all
+			return stream().count();
+		} 
+		// else: Ask directly
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.size();
+		}
+	}
+
+	
+	@Override
+	public void add(BlankNodeOrIRI graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource context = (Resource) rdf4jTermFactory.asValue(graphName);
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.add(subj, pred, obj,  context);
+			conn.commit();
+		}
+	}
+
+	@Override
+	public boolean contains(Optional<BlankNodeOrIRI> graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		Resource[] contexts = asContexts(graphName);
+		
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.hasStatement(subj, pred, obj, includeInferred, contexts);
+		}
+	}
+
+	private Resource[] asContexts(Optional<BlankNodeOrIRI> graphName) {
+		Resource[] contexts;
+		if (graphName == null) {
+			// no contexts == any contexts
+			 contexts = new Resource[0];
+		} else {	
+			BlankNodeOrIRI g = graphName.orElse(null);
+			Resource context = (Resource) rdf4jTermFactory.asValue(g);
+			contexts = new Resource[] { context };
+		}
+		return contexts;
+	}
+
+	@Override
+	public void remove(Optional<BlankNodeOrIRI> graphName, BlankNodeOrIRI subject, IRI predicate, RDFTerm object) {
+		Resource subj = (Resource) rdf4jTermFactory.asValue(subject);
+		org.eclipse.rdf4j.model.IRI pred = (org.eclipse.rdf4j.model.IRI) rdf4jTermFactory.asValue(predicate);
+		Value obj = rdf4jTermFactory.asValue(object);
+		Resource[] contexts = asContexts(graphName);
+
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(subj, pred, obj, contexts);
+			conn.commit();
```

Comment:
I am not sure how to do it efficiently without explicit begin/end in the Graph/Dataset model. However, as it stands, this method will not be useful for any but the most trivial cases, given the potential overhead for commit each time so it needs work to get into circulation.

If you added begin/end then you could also optimise to reduce the number of calls to getRepositoryConnection that may be resource or time intensive depending on the Repository implementation.